metricbeat/module/kubernetes/state_deployment/state_deployment_test.go

```diff
@@ -44,35 +44,76 @@ func TestEventMapping(t *testing.T) {
 	events, err := f.Fetch()
 	assert.NoError(t, err)
 
-	assert.Equal(t, 4, len(events), "Wrong number of returned events")
-
-	testCases := map[string]interface{}{
-		"_module.namespace": "default",
-
-		"name":   "jumpy-owl-redis",
-		"paused": false,
-
-		"replicas.available":   0,
-		"replicas.desired":     1,
-		"replicas.unavailable": 1,
-		"replicas.updated":     1,
-	}
+	assert.Equal(t, 5, len(events), "Wrong number of returned events")
 
+	testCases := testCases()
 	for _, event := range events {
 		name, err := event.GetValue("name")
-		if err == nil && name == "jumpy-owl-redis" {
-			for k, v := range testCases {
-				testValue(t, event, k, v)
+		if err == nil {
+			namespace, err := event.GetValue("_module.namespace")
+			if err == nil {
+				eventKey := namespace.(string) + "@" + name.(string)
```

Comment:
And I assume this one means an `@` in the field key? Not that ES would disallow this, but I wonder if it could cause issues on the query side or KB?

Reply:
This @ works in the same way, it's used to fetch the correct test case from here: https://github.com/elastic/beats/pull/6294/files/b514a509da8360ab225e85eb73df79468057cbd7#diff-2b3ff3319749ebeecd269e64715d1969R80

The @ is not in the final event